### DISCUSSION TITLE
How to add custom middleware for restricting different user type in filament?

### DISCUSSION THREAD

**Discussion by CodeWithPranta** [filamentphp/filament#8679] [General]
I'm building a school management system in filament3.0 I want to create different panel for admin, teacher, student and patient. But how to add custorm middleware in that case?

**Comment by krekas** [comment]
Just add a check in the `canAccessFilament()` method

**Reply by CodeWithPranta** [reply]
In User Model? Or inside PanelProvider?

**Reply by krekas** [reply]
Read docs https://filamentphp.com/docs/3.x/panels/installation#deploying-to-production

**Reply by CodeWithPranta** [reply]
` public function canAccessPanel(Panel $panel): bool
    {
        return str_ends_with($this->email, '@yourdomain.com') && $this->hasVerifiedEmail();
    }`
I saw the documentation. Suppose I set here if user_type == 1 then it's accessible. I'm writing this function inside User Model. So when I try to create a different panel for teacher where I write it ? Because I want only user_type == 2 user will be able to enter on my teacher panel.

**Reply by krekas** [reply]
```php
public function canAccessPanel(Panel $panel): bool
{
    if ($panel->getId() === 'teacher' && $this->user->user_type === 2)  {
        return true;
    }

    return false
}
```

Also use proper markdown to write code please. And I think you might need some kind of permissions, maybe even roles so better use them instead of `user_type`.
And think if someone would need to work on your code he should think what is that user type 2.

**Reply by CodeWithPranta** [reply]
Thanks. One thing, I want to keep default admin panel as well as different panel for student, teacher and parent. Can I add more condition inside canAccessPanel()?

**Reply by krekas** [reply]
Why wouldn't you be able to add? Add as many as you need

**Reply by CodeWithPranta** [reply]
Now I understand, thank you so much.
